File: LineUp/exit.py
from .Interpreter.line import Line


class Exit:
    def __init__(self, interpreter):
        self.interpreter = interpreter

    def load_command(self, command_name, *parameters):
        if type(command_name) is str:
            print(self.interpreter.global_variable[command_name].load_command(*parameters))
        else:
            print(command_name.load_command(*parameters))

File: php/LineUp/root_command/Exit.php
<?php
/**
 * Description of Exit
 *
 * @author Administrator
 */
class ExitUp {
    public $interpreter;
    public $e_type;

    public function __construct(&$interpreter, $e_type) {
        $this->interpreter = $interpreter;
        $this->e_type = $e_type;
    }
    
    public function load_command($command_name, $parameters) {
        if (is_string($command_name)) {
            $ex = array_splice($parameters, 1);
            $com = $this->interpreter->global_variable[$command_name]->load_command($parameters[0], $ex);
        } else {
            $com = $command_name->load_command($parameters[0], array_splice($parameters, 1));
        }
        if ($this->e_type == "print") {
            var_dump($com);
        } else {
            return array("/L_e*/", $com);
        }
    }
}
